services/src/main/java/org/exoplatform/officeonline/WOPIService.java

```diff
@@ -935,24 +952,25 @@ public String putSuggestedFile(EditorConfig config, String target, InputStream d
    * @throws PermissionDeniedException the permission denied exception
    */
   public String putRelativeFile(EditorConfig config,
-                              String filename,
-                              boolean overwrite,
-                              InputStream data) throws IllegalFileNameException,
-                                                FileNotFoundException,
-                                                RepositoryException,
-                                                FileLockedException,
-                                                UpdateConflictException,
-                                                FileExtensionNotFoundException,
-                                                PermissionDeniedException {
+                                String filename,
+                                boolean overwrite,
+                                InputStream data) throws IllegalFileNameException,
+                                                  FileNotFoundException,
+                                                  RepositoryException,
+                                                  FileLockedException,
+                                                  UpdateConflictException,
+                                                  FileExtensionNotFoundException,
+                                                  PermissionDeniedException {
 
     if (config.getPermissions().contains(Permissions.USER_CAN_NOT_WRITE_RELATIVE)) {
       throw new PermissionDeniedException("User " + config.getUserId() + " cannot write relative");
     }
 
     filename = Text.escapeIllegalJcrChars(filename);
-    // Check and escape newTitle
-    if (StringUtils.isBlank(filename)) {
-      throw new IllegalFileNameException("Provided filename is illegal");
+    
+    // TODO: hadnle long filenames by trimming it. Remember about the uniqueness.
```

Comment:
Add more details about uniqueness: a trimmed node name should not conflict with another node name starting with the same prefix but being truncated before an unique part. Tell also that eXo JCR limit is 512 characters.

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878